runtime/runtime/src/actions.rs

```diff
@@ -41,6 +39,27 @@ use near_vm_runner::logic::types::PromiseResult;
 use near_vm_runner::logic::{VMContext, VMOutcome};
 use near_vm_runner::precompile_contract;
 use near_vm_runner::ContractCode;
+use near_wallet_contract::wallet_contract;
+
+/// Returns `ContractCode` (if exists) for the given `account` or returns `StorageError`.
+/// For ETH-implicit accounts returns `Wallet Contract` implementation that it is a part
+/// of the protocol and it's cached in memory.
+fn get_contract_code(
+    runtime_ext: &RuntimeExt,
+    account: &Account,
+    protocol_version: ProtocolVersion,
+) -> Result<Option<ContractCode>, StorageError> {
+    let account_id = runtime_ext.account_id();
+    let code_hash = account.code_hash();
+    if checked_feature!("stable", EthImplicitAccounts, protocol_version)
+        && account_id.get_account_type() == AccountType::EthImplicitAccount
+    {
+        let contract = wallet_contract();
+        debug_assert!(code_hash == *contract.hash());
+        return Ok(Some(contract.clone()));
```

Comment:
I guess this is OK for now, as it’s unstable and the discussion on NEP-518 has not stabilized yet, but I do think that we’ll want something more stable, for users, than just "literally any eth implicit account lives with the latest version of the eth contract"

Also, you seem to have been careful to have an eth account be zero-balance-ready. Is there any use case for it? If yes, we should probably add that as a test, as currently the only test does have 1 yoctoNEAR and thus was already created before entering this function.

Disclaimer: I’m saying this because I’m not sure how the code would behave if you tried to call a function on the wallet contract without having funded the account first

Reply:
I think we should assume that Eth implicit accounts be have the same a the current Near implicit accounts. I think the behaviour for these is to reject `FunctionCall` actions if the account does not exist, but `Transfer` actions can create the account. `Transfer { deposit: 0 }` is allowed with zero-balance accounts so you can create implicit accounts for free.

Reply:
`meta_tx_create_implicit_account` test shows that we can create eth-implicit account with  `Transfer { deposit: 0 }`.
I've just added a test (`meta_tx_call_wallet_contract_account_does_not_exist`) that shows that `FunctionCall` action is rejected if the account does not exist.